dydxprotocol/v4-web
# Use abacus for simulate transactions and balance polling

Use abacus for simulate transactions and balance polling.



## Views

* `src/views/forms/AccountManagementForms/WithdrawForm.tsx`
  * Remove simulate Withdraw logic, now handled via abacus

* `src/views/forms/TransferForm.tsx`
  * Remove logic around fees as that is handled via abacus
  * Update abacus token value when asset changes, this is needed for fee calculation

## Constants/Types

* `src/constants/wallets.ts`
  * `DydxChainAsset` values needed to be changed to work with abacus, although a follow up would be to have the values defined in side abacus and then imported here.
  

## Functions

* `src/lib/abacus/dydxChainTransactions.ts`
  * add support for `TransactionType.simulateWithdraw` 
  * add support for `TransactionType.simulateTransferNativeToken` 
  * add support for `QueryType.GetAccountBalances` 

* `src/lib/abacus/stateNotification.ts`
  * Support `Changes.accountBalances`
 
## Hooks

* `src/hooks/useAccountBalance.ts`
  * Use balances from redux that is updated via abacus
  
* `src/hooks/useSubaccount.tsx`
  * Use usdc balance from redux that is updated via abacus
  * Remove simulate transaction functions, as they are handled via the abacus transaction protocol

## State

* `src/state/account.ts`
  * Save abacus balances to `balances`

## Packages

* `abacus`
  * updated: 0.5.0 -> 0.5.4


---

Testing:
- ensure transfers still work and show fees
- ensure withdraws work
- ensure auto deposit still works
